metagraph/src/annotation/taxonomy/tax_classifier.cpp

```diff
@@ -0,0 +1,300 @@
+#include "tax_classifier.hpp"
+
+#include <string>
+#include <vector>
+
+#include "annotation/representation/annotation_matrix/annotation_matrix.hpp"
+#include "common/unix_tools.hpp"
+#include "common/utils/string_utils.hpp"
+#include "common/logger.hpp"
+
+namespace mtg {
+namespace annot {
+
+using mtg::common::logger;
+
+bool TaxonomyBase::get_taxid_from_label(const std::string &label, TaxId *taxid) const {
+    if (label_type_ == TAXID) {
+        *taxid = std::stoul(utils::split_string(label, "|")[1]);
+        return true;
+    } else if (label_type_ == GEN_BANK) {
+        std::string acc_version = get_accession_version_from_label(label);
+        auto it = accversion_to_taxid_map_.find(acc_version);
+        if (it == accversion_to_taxid_map_.end()) {
+            return false;
+        }
+        *taxid = it->second;
+        return true;
+    }
+
+    logger->error("Error: Could not get the taxid for label {}", label);
+    exit(1);
+}
```

Comment:
where is this called?

Reply:
`get_taxid_from_label` will be used in the next PR. I implemented the entire `.hpp` file in this PR  and wanted to add a reasoning about why/how to use the `enum LabelType`